File: codegen/results/v001/apple-foundationmodel/20260409_105028/python/004_json_filter/solution.py
import json

def filter_and_sort_records(input_file='input/data.json', output_file='output.json'):
    # Read data from the input file
    with open(input_file, 'r') as file:
        data = json.load(file)

    # Filter records based on the criteria: active=True and age>=30
    filtered_records = [record for record in data if record['active'] and record['age'] >= 30]

    # Sort the filtered records by name in ascending order
    filtered_records.sort(key=lambda x: x['name'])

    # Convert the filtered and sorted list back to JSON format
    output_data = json.dumps(filtered_records, indent=4)

    # Output the result to stdout
    print(output_data)

# Call the function
filter_and_sort_records()

File: codegen/results/v001/apple-foundationmodel/20260409_105028/typescript/004_json_filter/solution.ts
import fs from 'fs';
import path from 'path';

async function main() {
    const inputPath = path.join(__dirname, 'input', 'data.json');
    const outputPath = path.join(__dirname, 'output', 'filtered_data.json');

    // Read the input file
    const inputData = await fs.promises.readFile(inputPath, 'utf8');

    // Parse the input data
    const jsonData = JSON.parse(inputData);

    // Filter records where 'active' is true and 'age' is 30 or older
    const filteredData = jsonData.filter(record => record.active && record.age >= 30);

    // Sort the filtered records by 'name' in ascending order
    const sortedData = filteredData.sort((a, b) => a.name.localeCompare(b.name));

    // Write the output to a new file
    await fs.promises.writeFile(outputPath, JSON.stringify(sortedData, null, 2));

    console.log('Filtered and sorted data output written to:', outputPath);
}

main().catch((error) => {
    console.error('An error occurred:', error);
});